pkg/compact/compact.go

```diff
@@ -743,6 +746,46 @@ func (cg *Group) areBlocksOverlapping(include *metadata.Meta, excludeDirs ...str
 	return nil
 }
 
+// isEnoughDiskSpace checks with a heuristic if there is enough disk space in the compaction group
+// in the specified directory. It adds up the space requirements of all the blocks in the compaction group
+// and then adds the smallest one again. The specified directory needs to have at least that amount of available bytes.
+// Doing this check prevents (in most cases) from downloading a lot of data and only then finding out that there is not
+// enough space.
+func (cg *Group) isEnoughDiskSpace(ctx context.Context, dir string) error {
```

Comment:
Do you think it might be more idiomatic to return `(bool, error)`? This way you can differentiate between a failure to calculate and a negative response? You could also do that with a specially typed err. Either way, doesn’t matter much for this use case

Reply:
I thought about this and arrived at exactly the same conclusion - it is unnecessary in this case unless we will want to add some kind of flag to skip this check. (: